src/main/kotlin/gg/skytils/skytilsmod/utils/MojangUtil.kt

```diff
@@ -107,6 +162,19 @@ object MojangUtil {
         fixedRateTimer("Mojang-Fake-Requests-Insert", startAt = Date((System.currentTimeMillis() / 60000) * 60000 + 48000), period = 60_000L) {
             requestCount.set(0)
         }
+
+        Skytils.IO.launch {
+            val usernames = ArrayDeque<Pair<String, Long>>(10)
+            usernameQueue.collect { name ->
+                usernames.add(name)
+                if (usernames.size >= 10 || usernames.isNotEmpty() && System.currentTimeMillis() - usernames.first().second >= 3000L) {
```

Comment:
this condition good?

Reply:
If you're trying to create what is effectively a bucket ratelimiter, there are probably better ways to handle this (see [Levelhead](https://github.com/Sk1erLLC/Levelhead/blob/master/src/main/kotlin/club/sk1er/mods/levelhead/core/RateLimiter.kt)). Otherwise, you'd probably want your `usernameQueue` flow to have a replay cache (and buffer) of 10 so you can access queued values more than once. 
Also, It'd probably be nice to add qualifying parenthesis around the second condition because it's not very straightforward from a glance.